### ISSUE TITLE
gem install fails with "can't modify frozen String" for certain gems

### ISSUE THREAD

**Issue by mojavelinux** [rubygems/rubygems#726]
For certain gems, the `gem install` command fails with the following error:

```
RuntimeError: can't modify frozen String
```

I've traced this back to the following line in _rubygems/installer.rb_:

``` ruby
def generate_bin
  ...
  spec.executables.each do |filename|
  filename.untaint # (1)
  ...
end
```

(1) Failing line

This seems to only happen when the executables in the spec are defined as follows:

``` ruby
s.executables = s.files.grep(/^bin\//) { |f| File.basename f }
```

Any attempt to duplicate or dynamically recreate the executables array yields the same failure.

``` ruby
s.executables = s.files.grep(/^bin\//) { |f| File.basename f }
s.executables = s.executables.map {|f| f.dup }
puts s.executables.first.frozen? # (1)
```

(1) Prints "false"

I've checked to see if the string is frozen at the time the specification is loaded and it isn't.

If the executable names are defined using a regular string array, then it works:

``` ruby
s.executables = ['name-of-executable']
```

Ruby version: 2.1.0.preview2
RubyGems version: 2.1.11

**Comment by pboling** [comment]
Apparently this is still happening on Ruby 2.3.3 and 2.4.0 installing the gem rainbow, version 2.2.1.

https://travis-ci.org/pboling/sanitize_email/jobs/202663386
https://travis-ci.org/pboling/sanitize_email/jobs/202663390

**Comment by segiddins** [comment]
That's a new bug introduced in 2.6.x and fixed in 2.6.10
